draft-ietf-jose-hpke-encrypt.md

```diff
@@ -106,167 +109,237 @@ informative:
+~~~
+{: #ciphersuite-int-algs title="Algorithms using HPKE for Integrated Encryption"}
+
+The HPKE KEM, KDF, and AEAD values are chosen from the IANA HPKE registry {{IANA.HPKE}}.
+
+## JWE Compact Serialization Example {#compact-example}
 
-Below is an example of a Compact JWE using HPKE integrated encryption:
+Below is an example of a JWE using the Compact Serialization and Integrated Encryption with HPKE:
 
 ~~~
-{::include-fold examples/compact_example.txt}
+eyJhbGciOiJIUEtFLTAiLCJraWQiOiJ5Q25mYm1ZTVpjV3JLRHRfRGpOZWJSQ0IxdnhWb3F2NHVtSjRXSzhSWWprIn0.BLAJX8adrFsDKaoJAc3iy2dq-6jEH3Uv-bSgqIoDeREqpWglMoTS67XsXere1ZYxiQKEFU6MbWe8O7vmdlSmcUk..NcN9ew5aijn8W7piLVRU8r2cOP0JKqxOF4RllVsJM4qsAfVXW5Ka6so9zdUmXXNOXyCEk0wV_s8ICAnD4LbRa5TkhTeuhijIfAt9bQ2fMLOeyed3WyArs8yaMraa9Zbh4i6SaHunM7jU_xoz_N2WbykSOSySmCO49H4mP3jLW9L_TYQfeVfYsrB8clqokZ8h-3eQGNwmOPtkjWdpAfaHUsp4-HC9nRd6yrTU6mV65Nn2iYynu3Xkgy2Lm-kQKDavIEW3PBpEeiw6mtPJE9o8sT-0lZ9kpWtqog2XbNGEfjSOjujvNe1b0g4-FdNFMFO_fo0rxe902W1pGT7znv4Q-xBkIydK4ZwjiFN6dAXutnococ37A0Hr5esPLwHRTTrBFw.
 ~~~
 
-The keys used for this example are in {{keys-used}}.
+The key used for this example is in {{int-key}}.
+
+## Flattened JWE JSON Serialization Example {#flattened-example}
+
+Below is an example of a JWE using the Flattened JSON Serialization and Integrated Encryption with HPKE:
+
+~~~
+{
+  "ciphertext": "LabI8_KIPDbymUSbyVctj8AfISXQ07sMt1xQ1lrS-0heU2jjejpQIK75K1KXcvwn15E6Kil_tJ6LBcYCu02O1H8_aooJGuoLw1vEzQn16h498YX9e2SA2IcVrJTkcCjL7YpF9fsAF3JEzGfsmmrpZPPVdxCn7g8dkGRcyulnHrNvBu4BFtub-URtf-nYCFIJHZ4k-ul9fDddquicFzCxQonx66-ZX5nbj6azHG65tAZntd6VFkRgihdxTvIpvTS4gfulQeKyShbiw-OCJNbzFdEnOKEMnsyqRjwG7iVrFEilFAMsvLJ14-lcuR5btIkUntIwlnsfUa2Ytk33znCfAFN0wYukdDvJe-V0nnNUFlOeLyYV0eEGisgC9dQQ1kFu3g",
+  "encrypted_key": "BAOlZ-VnbhQu4NOlTlDAVYwUJB-Q6YcWwnRNWK6YLSiHHlW4rN0qUzBJ3Rc2_y8nkasn8nUVGBzdq7OhdKKiLq4",
+  "aad": "VGhlIEZlbGxvd3NoaXAgb2YgdGhlIFJpbmc",
+  "protected": "eyJhbGciOiJIUEtFLTAiLCJraWQiOiJ5Q25mYm1ZTVpjV3JLRHRfRGpOZWJSQ0IxdnhWb3F2NHVtSjRXSzhSWWprIn0"
+}
+~~~
+
+The key used for this example is in {{int-key}}.
 
 # Key Encryption
 
 When using the JWE JSON Serialization,
-recipients using JOSE-HPKE can be added alongside other recipients
+recipients using Key Encryption with HPKE can be added alongside other recipients
 (e.g., those using `ECDH-ES+A128KW` or `RSA-OAEP-256`),
-since HPKE is used to encrypt the Content Encryption Key,
-which is then processed as specified in JWE.
-
-The encoding of the protected header remains consistent with existing JWE rules.
+since HPKE is used to encrypt the Content Encryption Key (CEK).
 
-In HPKE JWE Key Encryption:
+When using Key Encryption with HPKE:
 
-- The Key Management Mode is Key Encryption.
-- When all recipients use the same HPKE algorithm to secure the Content Encryption Key, the JWE Protected Header SHOULD contain "alg".
+- When all recipients use the same JWE algorithm to secure the Content Encryption Key, the JWE Protected Header SHOULD contain "alg".
```

Comment:
I don't see the point behind this SHOULD requirement. When all recipients use the same "alg", in this case HPKE Key Encryption, then the chosen HPKE CipherSuite is tied to the resulting ciphertext / CEK anyway.

Reply:
We can say something like:
When all recipients use the same JWE algorithm to secure the Content Encryption Key, placing the "alg" parameter in the JWE Protected Header avoids unnecessary repetition in the per-recipient headers.